analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.cs

```diff
@@ -88,3 +88,37 @@ internal class PrivateClass2
         }
     }
 }
+
+namespace TestsFromDeprecatedS4214
+{
+    public class Program
+    {
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        public static extern bool RemoveDirectory1(string name); // Noncompliant {{Make this native method private and provide a wrapper.}}
+        //                        ^^^^^^^^^^^^^^^^
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        protected static extern bool RemoveDirectory2(string name); // Noncompliant
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        internal protected static extern bool RemoveDirectory3(string name); // Noncompliant
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        internal static extern bool RemoveDirectory4(string name);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        private static extern bool RemoveDirectory5(string name);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)] // Error [CS0601]
+        public extern bool RemoveDirectory6(string name);      // Noncompliant
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)] // Error [CS0601]
+        public static bool RemoveDirectory7(string name); // Error [CS0501] - so do not raise an issue
+    }
+
+    internal class Foo
+    {
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        public static extern bool RemoveDirectory1(string name); // Compliant because effective accessibility is not public
+    }
+}
```

Comment:
Copy of https://github.com/SonarSource/sonar-dotnet/blob/c18ed76dbbc997007e41be9e9e49df3aa68ab233/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PInvokesShouldNotBeVisible.cs#L1-L36